tests/test_dsp.py

```diff
@@ -541,3 +541,74 @@ def test_minimum_phase():
 
     assert imp_minphase.n_samples == imp_linphase.n_samples
     npt.assert_allclose(imp_minphase.time, ref.time)
+
+
+def test_convolve_default():
+    x = pf.Signal([1, 0.5, 0.25, 0], 44100)
+    y = pf.Signal([1, -1, 0], 44100)
+
+    res = dsp.convolve(x, y)
+    desired = np.array([[1, -0.5, -0.25, -0.25, 0, 0]])
+    np.testing.assert_allclose(res.time, desired, atol=1e-10)
+
+
+def test_convolve_sampling_rate_error():
+    x = pf.Signal([1, 0.5, 0.25, 0], 44100)
+    y = pf.Signal([1, 0.5, 0.25, 0], 48000)
+
+    with pytest.raises(ValueError, match="sampling rates"):
+        dsp.convolve(x, y)
+
+
+def test_convolve_fft_norm_error():
+    x = pf.Signal([1, 0.5, 0.25, 0], 44100, fft_norm='unitary')
+    y = pf.Signal([1, 0.5, 0.25, 0], 44100, fft_norm='amplitude')
+
+    with pytest.raises(ValueError, match="FFT norms"):
+        dsp.convolve(x, y)
+
+
+@pytest.mark.parametrize("method", ['overlap_add', 'fft'])
+def test_convolve_mode_full(method):
+    x = pf.Signal([1, 0.5, 0.5, 0.1], 44100)
+    y = pf.Signal([1, -1, 0.1], 44100)
+
+    res = dsp.convolve(x, y, mode='full', method=method)
+    desired = np.array([[1, -0.5, 0.1, -0.35, -0.05, 0.01]])
+    np.testing.assert_allclose(res.time, desired, atol=1e-10)
+
+
+@pytest.mark.parametrize("method", ['overlap_add', 'fft'])
+def test_convolve_mode_cut(method):
+    x = pf.Signal([1, 0.5, 0.5, 0.1], 44100)
+    y = pf.Signal([1, -1, 0.1], 44100)
+
+    res = dsp.convolve(x, y, mode='cut', method=method)
+    desired = np.array([[1, -0.5, 0.1, -0.35]])
+    np.testing.assert_allclose(res.time, desired, atol=1e-10)
+
+
+@pytest.mark.parametrize("method", ['overlap_add', 'fft'])
+def test_convolve_mode_cyclic(method):
+    x = pf.Signal([1, 0.5, 0.5, 0.1], 44100)
+    y = pf.Signal([1, -1, 0.1], 44100)
+
+    res = dsp.convolve(x, y, mode='cyclic', method=method)
+    desired = np.array([[0.95, -0.49, 0.1, -0.35]])
+    np.testing.assert_allclose(res.time, desired, atol=1e-10)
```

Comment:
I think you could make this one test with a nested use of parameterize. This is fancy and a student of ours showed it to me:
``` python
@pytest.mark.parametrize("method", ['overlap_add', 'fft'])
@pytest.mark.parametrize("mode,desired", [('full', [desired result]), ('cut', [desired result]), ('cyclic', [desired result]))
def test_convolve_mode_and_method(method, mode, desired):
     # out the test here
```